src/js/builtins/CommonJS.ts

```diff
@@ -369,6 +369,102 @@ export function createRequireCache() {
   return proxy;
 }
 
+type ExtensionChange = (extension: string, kind: number) => void;
+$visibility = "Private";
+export function createExtensionsObject(onChange: ExtensionChange) {
+  // These function bodies are only called if directly called, re-assigning one
+  // of these to `require.extensions` will trigger the native loader, bypassing
+  // all JS. The implementations are not complete, since it isn't even known
+  // that any production code calls them directly.
+  function loadJS(module, filename) {
+    // @ts-expect-error `require` is not a same-scope reference
+    let content = require("node:fs").readFileSync(filename, "utf8");
+    if (content.charCodeAt(0) === 0xFEFF) {
+      content = content.slice(1);
+    }
+    module._compile(content, filename, "js");
+  }
+  function loadJSON(module, filename) {
+    // @ts-expect-error `require` is not a same-scope reference
+    let content = require("node:fs").readFileSync(filename, "utf8");
+    if (content.charCodeAt(0) === 0xFEFF) {
+      content = content.slice(1);
+    }
+    try {
+      $putByIdDirect(module, 'exports', JSON.parse(content));
+    } catch (err: any) {
+      err.message = filename + ': ' + err.message;
+      throw err;
+    }
+  }
+  function loadNodeAddon(module, filename) {
+    // @ts-expect-error `require` is not a same-scope reference
+    return process.dlopen(module, process.platform === 'win32' ? require('node:path').toNamespacedPath(filename) : filename);
+  }
+  let t: import("bun").Transpiler;
+  function loadTS(module, filename) {
+    // @ts-expect-error `require` is not a same-scope reference
+    let content = require("node:fs").readFileSync(filename, "utf8");
+    if (content.charCodeAt(0) === 0xFEFF) {
+      content = content.slice(1);
+    }
+    t ??= new Bun.Transpiler();
```

Comment:
do not use this

Reply:
this should become a virtual module when calling into the code in module_loader.zig, passing the filename and  source code from fs.readFileSync as  arguments